### ISSUE TITLE
Add user with role and passowrd

### ISSUE THREAD

**Issue by sheikh303** [dwijitsolutions/laraadmin#183]
Can not find option to add new user under user module.
Although can add employee but they are not showing under user module with authentication.

**Comment by azarzaza** [comment]
u can add user in employe module  because in laraadmin employee module work
like user module

**Comment by sheikh303** [comment]
Thanks a lot for the quickest response :)

Added few employees, Now I want them to login to the backend. but since
they are not users table they dont have password nor received any
invitation email to join the system.

Thanks & Regards,
----
*Sheikh Faiyaz Moorsalin*
Assistant General Manager
ERGO VENTURES PVT. LTD.  <http://www.ergo-ventures.com/>
+8801938855538

Skype: sheikh303

**Comment by azarzaza** [comment]
please configure your mail address then only you received the password
through mail

**Comment by sheikh303** [comment]
Yes. mail is configured. I can receive email when I try forgot password.
But not received any invitation email after creating a employee user.

Thanks & Regards,
----
*Sheikh Faiyaz Moorsalin*
Assistant General Manager
ERGO VENTURES PVT. LTD.  <http://www.ergo-ventures.com/>
+8801938855538

Skype: sheikh303

**Comment by azarzaza** [comment]
you can check the employee controllers does the mail is correctly
configured

**Comment by sheikh303** [comment]
/**
     * Store a newly created employee in database.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return \Illuminate\Http\Response
     */
    public function store(Request $request)
    {
        if(Module::hasAccess("Employees", "create")) {

            $rules = Module::validateRules("Employees", $request);

            $validator = Validator::make($request->all(), $rules);

            if ($validator->fails()) {
                return
redirect()->back()->withErrors($validator)->withInput();
            }

            $insert_id = Module::insert("Employees", $request);

            return redirect()->route(config('laraadmin.adminRoute') .
'.employees.index');

        } else {
            return redirect(config('laraadmin.adminRoute')."/");
        }
    }


Not seeing any email trigger in the method or any storing in user's module
Please suggest.

**Comment by azarzaza** [comment]
which version of laravel do u using

**Comment by sheikh303** [comment]
clone it directly from your repo. few days ago.
will check and confirm you in few hours

**Comment by azarzaza** [comment]
please work with fresh laraadmin that should be fine for working

**Comment by azarzaza** [comment]
i think ur working with old one latest one having email send feature

**Comment by sheikh303** [comment]
could you please guide me to update to latest one
I followed the installation guide accordingly

**Comment by azarzaza** [comment]
Use the composer to install and for ur work use 5.2 laravel version

**Comment by sheikh303** [comment]
Just checked. My laravel version is 5.2.45.

Thanks & Regards,
----
*Sheikh Faiyaz Moorsalin*
Assistant General Manager
ERGO VENTURES PVT. LTD.  <http://www.ergo-ventures.com/>
+8801938855538

Skype: sheikh303

**Comment by azarzaza** [comment]
K use it composer to install laravel 5.2 then use composer require to add
the laraadmin package. Then add the laraadmin class to config file

**Comment by azarzaza** [comment]
Please refer laraadmin document

**Comment by sheikh303** [comment]
I followed the steps exactly mentioned in the following documentation. yet
didn't get the functionality as you mentioned for employee module.

   1. http://laraadmin.com/docs/1.0/installation
   2. http://laraadmin.com/docs/1.0/upgrades


Thanks & Regards,
----
*Sheikh Faiyaz Moorsalin*
Assistant General Manager
ERGO VENTURES PVT. LTD.  <http://www.ergo-ventures.com/>
+8801938855538

Skype: sheikh303

**Comment by azarzaza** [comment]
Over here mail functionality working

**Comment by azarzaza** [comment]
Send me the snap of employees controller store function

**Comment by sheikh303** [comment]
Employees and Users controllers are attached. Please find in attached

Thanks & Regards,
----
*Sheikh Faiyaz Moorsalin*
Assistant General Manager
ERGO VENTURES PVT. LTD.  <http://www.ergo-ventures.com/>
+8801938855538

Skype: sheikh303

**Comment by Martenvanurk** [comment]
I'm using rap2hpoutre/create-user-command at the command line for this.

**Comment by becology1100** [comment]
I have the same issue, i cant update user password .. any updates?

**Comment by bobbysiagian** [comment]
tested and looks from my side @becology1100

**Comment by becology1100** [comment]
What is the solution should i test ? 
I have no button to update users and in employe page i have no fields to update password too ?
do you got a solution for that ?
thanks :)

**Comment by bobbysiagian** [comment]
have you see this menu ?
update password in users detail, from tab "Account Setting"

[https://s26.postimg.org/ju8l9c8fd/Screen_Shot_2017-09-23_at_4.19.11_PM.png](url)

**Comment by becology1100** [comment]
NO, only time line and general info

**Comment by bobbysiagian** [comment]
I think there is something wrong on your side.
Pls recheck you laraadmin. @becology1100

**Comment by wadleo** [comment]
I think there should be a way to directly created users from the users module

**Comment by wadleo** [comment]
but since I have access to my database, I will just replicate the first account

**Comment by leopinzon** [comment]
@sheikh303 I faced the same problem and found the issue (in my case) at the line 114 from the EmployeesController.php: 

```
if(env('MAIL_USERNAME') != null && env('MAIL_USERNAME') != "null" && env('MAIL_USERNAME') != "") {
    // Send mail to User his Password
    Mail::send('emails.send_login_cred', ['user' => $user, 'password' => $password], function ($m) use ($user) {
    $m->from('hello@laraadmin.com', 'LaraAdmin');
    $m->to($user->email, $user->name)->subject('LaraAdmin - Your Login Credentials');
});
} else {
    Log::info("User created: username: ".$user->email." Password: ".$password);
}
```
If you are not using username in order to use your SMTP service, the notification sent to the user will be overlooked. 

Hope this helps, 
Best

**Comment by benedict-payot** [comment]
This issue has been for a while now and I've encountered it myself. Here's what I did:

Turns out, env('MAIL_USERNAME') is not being read on the controller so I assigned $_ENV['MAIL_USERNAME'] to a variable and replaced all the instances of env('MAIL_USERNAME').

Here's my final line 113-115:

```
$mail_username = $_ENV['MAIL_USERNAME'];
			
if($mail_username != null && $mail_username != "null" && $mail_username != "") {
```

Hope this helps!

**Comment by Muhammadinaam** [comment]
Hi, this is a nice package but I think that the developers have stopped working on it. Maybe you can try this package: https://github.com/Muhammadinaam/speed-admin. It supports Laravel 8+.
